packages/nimble-components/src/select/specs/IxD.md

```diff
@@ -0,0 +1,254 @@
+# Nimble Select (IxD)
+
+## Overview
+The select component is used to present a pre-defined list of options that end-users can choose from. Selects simplify choice-making, reduce errors, and conserve screen space for more efficient and user-friendly interactions.
+
+> NOTE: THE VISUAL DESIGN IN THIS DOCUMENT MAY NOT BE ACCURATE
+
+### Background
+- [Select and Combobox Refresh HLD (#1303)](https://github.com/ni/nimble/issues/1303)
+- [ViD Select Styles (Figma)](https://www.figma.com/file/PO9mFOu5BCl8aJvFchEeuN/Nimble_Components?type=design&node-id=2597-83231&mode=design)
+
+## Usage
+When to use:
+- To present a pre-defined list of options
+- For filtering or sorting data in tables or lists
+- For selecting settings or preferences
+- For single selection
+
+When not to use:
+- To allow end-users to add custom options
+- When there are fewer than three options
+- For complex and/or hierarchical data
+- For multi selection
+
+### Anatomy
+![ ](spec-images/select-1.png)
+
+| Element | Description |
+| ---------- | ---------- |
+| Label | Text that tells the end-user what to expect in the list of options | 
+| Helper text | Text that provides extra context or instruction about what the end-user should select | 
+| Field button | Click area that opens the menu and displays the current selected option| 
+| Menu | A list of options to choose from displayed as an *open* state | 
+| Menu item | A selection the end-user can make, shown with other options in the menu | 
+
+#### Label
+Selects should always have a label, unless approved by an interaction designer or user researcher (usually in toolbar use cases).
+
+Keep the label text as concise as possible, limited to a single line of text. Use sentence casing.
+
+In a form, use the label to indicate whether the select is optional or required for the end-user to interact with. If most of the components in the form are required, indicate which components are optional. If most of the components in the form are optional, indicate which components are required. Required or optional indication in the label text should always follow the pattern "[Label] (optional)" or "[Label] (required)".
+
+![ ](spec-images/select-2.png)
+
+#### Helper Text
+> NOTE: NEEDS VISUAL DESIGN FOR TYPOGRAPHY
+
+Only include helper text to provide information that aids the end-user in choosing the right selection. Keep the helper text concise as possible, limited to a single line of text. Use sentence casing.
```

Comment:
Just for clarity, is the helper text meant to be user-configurable, or reserved for the `Select` component to use when the situation calls for it? For example, one idea that was floated was we could use this area to provide the group name for the current selection if groups were being used.

If, instead, the idea was that the helper text was user-configurable, I would think this API would be available on _all_ Nimble input components. This would concern me.

Reply:
Yes, I think it is a requirement that most input components have client-user configurable helper text. This is to provide additional contextual information in a way that can still be accessed on touch devices, where the end-user cannot hover for help text or tool tips.

I do not think we should use this area to indicate the group name.

What are your concerns?

Reply:
The primary concern I have with this idea of helper text is essentially adding unbounded height to the component. As you mention the helper text seems analogous to a tooltip, which oftentimes can span multiple lines. If the motivation for adding this is enabling the components on mobile devices to better supply contextual information, I think there are other options to consider. One is providing an interactive icon that could be slotted in the component, which a user could touch to bring up a tooltip.

The secondary concern is the helper text's overlapping of the error text (which maybe we should mentioning in this document?). While I think it may be _okay_ to have the error text trump the helper text, there are other patterns, like the one I mention above that allow for available helper text, even in the presence of an error.

Another consideration is that we have to reconcile the recommendation of how group names are rendered with the Combobox (which I noticed its IxD document does not outline, but it will need the same behaviors as the `Select`.). While @NIbokeefe did provide a design option to display the group name within the input field, I don't think this is the ideal choice for something like a Combobox, where I think we should reserve the majority of that space for the actual interactive input element. I'll note that Brandon did also provide a design for displaying the group name in the upper right of the component (next to the label), which might satisfy my concern, if we didn't want to display it in the error text location.